### PR TITLE
added support for charge 2024 feat with additional GWF 2014/24

### PR DESCRIPTION
**Pull request by dmportella**
updated settings
added support for 2014/24 gwf since the dmg now is addtional damage die.

# Old charger feat
Screenshot 2024-11-23 154444

# New Charger feat
Screenshot 2024-11-23 154521

# tests

## Great axe gwf 2014

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 155713
## Great axe gwf 2014

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 155720
## Great axe gwf 2024

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 155842
## polearms 2014 gwf 2024

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 155912
## polearms 2024 gwf 2024

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 155928
## polearms 2024 gwf 2014

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 160613
## polearms 2014 gwf 2014

> should apply gwf on charger feat dice because weapon is a heavy/two-handed weapon

Screenshot 2024-11-23 160551
## Non-heavy weapon

> should **NOT** apply gwf on charger feat dice because weapon is NOT a heavy/two-handed weapon

Screenshot 2024-11-23 160756

### CODE REVIEW

File: src/dndbeyond/content-scripts/character.js [hunk 1]
```diff
@@ -456,6 +456,26 @@ function handleSpecialMeleeAttacks(damages=[], damage_types=[], properties, sett
...
+            let charge_dmg = "1d8";
+            // apply GWF if needed
+            if(((properties["Attack Type"] == "Melee" && ((properties["Properties"].includes("Versatile") && character.getSetting("versatile-choice") != "one") || 
+                properties["Properties"].includes("Two-Handed"))) ||
+                action_name == "Polearm Master - Bonus Attack" ||
+                action_name == "Pole Strike")) {
+                if(character.hasGreatWeaponFighting(2014)) {
+                    charge_dmg += "ro<=2";
+                } else if(character.hasGreatWeaponFighting(2024)) {
+                    charge_dmg = charge_dmg.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:470]: You don't need all this complex regexp, you know `charge_dmg` is `1d8`, you can just do `charge_dmg = "1d8min3"` or use `+=` safely, like for 2014 version

**Review by kakaroto — COMMENTED** [review]
One small comment, looks good otherwise 👍

**Comment by dmportella** [comment]
@kakaroto change done <3

**Review by kakaroto — APPROVED** [review]
Thanks for the quick change!

**Pull request merged**